python/fastquant/strategies/base.py

```diff
@@ -46,14 +46,34 @@ class BaseStrategy(bt.Strategy):
             "execution_type",
             "close",
         ),  # Either open or close, to indicate if a purchase is executed based on the next open or close
-        ("periodic_logging", False),
+        ("periodic_logging", False), #DO NOT set to True else return_transactions
         ("transaction_logging", True),
     )
 
     def log(self, txt, dt=None):
+        """
+        FIXME: parsing data from txt assumes a format
+        """
         dt = dt or self.datas[0].datetime.date(0)
         print("%s, %s" % (dt.isoformat(), txt))
 
+        txts = txt.split(', ')
+        if ("BUY EXECUTED" in txts) | ("SELL EXECUTED" in txts):
+            if "BUY EXECUTED" in txts: 
+                #"BUY EXECUTED, Price: %.2f, Cost: %.2f, Comm: %.2f"
+                buy_price = txts[1].split(": ")[1]
+            else:
+                buy_price = None
+            if "SELL EXECUTED" in txts: 
+                #"SELL EXECUTED, Price: %.2f, Cost: %.2f, Comm: %.2f"
+                sell_price = txts[1].split(": ")[1]
+            else:
+                sell_price = None
+            cost = txts[2].split(": ")[1]
+            comm = txts[3].split(": ")[1]
+            data = (dt.isoformat(), buy_price, sell_price, cost, comm)
```

Comment:
@enzoampil We should also add 'returns' here, right?

Reply:
Ah the returns from the trade. Yeah let's figure out how to do this.

Reply:
Also realized now I think adding to transaction history should happen at `next` instead of `log` since only text is supposed to be passed here.

Reply:
Re: returns
We can just compute `returns` inside log and add it as an additional column.
I chose to parse log over next or notify_order because all important stuff is logged works as an easy catch filter. But I agree we should access data from `strat`.